src/adaptors/fs.js

```diff
@@ -0,0 +1,88 @@
+var concat = require('concat-stream')
+var fs = require('fs-blob-store')
+var lockFile = require('lockfile')
+
+function BlobStore (path) {
+  this.store = fs(path)
+  this.LOCK_PATH = path + '/repo.lock'
+}
+
+BlobStore.prototype = {
+  read: function (key, cb) {
+    var store = this.store
+    var LOCK_PATH = this.LOCK_PATH
+
+    function onFinish (buff) {
+      lockFile.unlock(LOCK_PATH, function (err) {
+        if (err) return cb(err)
+
+        cb(null, buff.toString('utf8'))
+      })
+    }
+
+    function onLock (err) {
+      if (err) return cb(err)
+
+      var rs = store.createReadStream(key)
+
+      rs.on('error', cb)
+      rs.pipe(concat(onFinish))
+    }
+
+    lockFile.lock(LOCK_PATH, {}, onLock)
+  },
+
+  readWithoutLock: function (key, cb) {
+    var rs = this.store.createReadStream(key)
+
+    rs.on('error', cb)
+    rs.pipe(concat(function (buff) {
+      cb(null, buff.toString('utf8'))
+    }))
+  },
+
+  append: function (key, content, cb) {
+    return this._write(key, content, 'a', cb)
+  },
+
+  write: function (key, content, cb) {
```

Comment:
good point, initially I was returning the writeable stream but it got lost when implementing the locks. Will create a richer API as I move forward.